tensorflow-stubs/__init__.pyi

```diff
@@ -35,97 +81,159 @@ class dtypes:
 # https://github.com/tensorflow/tensorflow/blob/28340a4b12e286fe14bb7ac08aebe325c3e150b4/tensorflow/python/ops/variables.py#L1565
+
+
 def global_variables_initializer() -> Any: ...
 
 # Original function definition for reset_default_graph here:
 # https://github.com/tensorflow/tensorflow/blob/28340a4b12e286fe14bb7ac08aebe325c3e150b4/tensorflow/python/framework/ops.py#L5531
+
+
 def reset_default_graph() -> Graph: ...
 
 
 # Original function definition for placeholder here:
 # https://github.com/tensorflow/tensorflow/blob/28340a4b12e286fe14bb7ac08aebe325c3e150b4/tensorflow/python/ops/array_ops.py#L1693
 # TODO: improve types
-def placeholder(dtype: Any, shape: Any = None, name: Optional[str] = None) -> Any: ...
+def placeholder(dtype: Any,
+                shape: Any=...,
+                name: Optional[str]=...
+                ) -> Any: ...
 
 # Original function definition for sparse_placeholder here:
 # https://github.com/tensorflow/tensorflow/blob/28340a4b12e286fe14bb7ac08aebe325c3e150b4/tensorflow/python/ops/array_ops.py#L1749
 # TODO: improve types
-def sparse_placeholder(dtype: Any, shape: Any = None, name: Optional[str] = None) -> Any: ...
+
+
+def sparse_placeholder(dtype: Any,
+                       shape: Any=...,
+                       name: Optional[str]=...
+                       ) -> Any: ...
 
 # Original function definition for sparse_tensor_to_dense here:
 # https://github.com/tensorflow/tensorflow/blob/d8f9538ab48e3c677aaf532769d29bc29a05b76e/tensorflow/python/ops/sparse_ops.py#L948
 # sp_input is SparseTensor
 # returns Tensor
-def sparse_tensor_to_dense(sp_input: Any, default_value: Any=0, validate_indices: bool=True, name: Optional[str]=None) -> Any: ...
+
+
+def sparse_tensor_to_dense(sp_input: Any,
+                           default_value: Any=...,
+                           validate_indices: bool=...,
+                           name: Optional[str]=...
+                           ) -> Any: ...
 
 # Original function definition for shape here:
 # https://github.com/tensorflow/tensorflow/blob/28340a4b12e286fe14bb7ac08aebe325c3e150b4/tensorflow/python/ops/array_ops.py#L197
 # TODO: improve types. return type of None here is a hack
 # input is `Tensor` or `SparseTensor`
 # out_type is an optional integral data-type (`int32` or `int64`).
 # returns a `Tensor` of type specified by `out_type`
-def shape(input: Any, name: Optional[str] = None, out_type: Any = None) -> Any: ...
+
+
+def shape(input: Any,
+          name: Optional[str]=...,
+          out_type: Any=...
+          ) -> Any: ...
 
 # Original function definition for truncated_normal here:
 # https://github.com/tensorflow/tensorflow/blob/70cd9ed2d2ea37a6da6f813a99b32c03e90736a4/tensorflow/python/ops/random_ops.py#L139
-def truncated_normal(shape: Any, mean: Any=0.0, stddev: Any=1.0, dtype: Any=dtypes.float32, seed: Any=None, name: Optional[str]=None) -> Any: ...
+
+
+def truncated_normal(shape: Any,
+                     mean: Any=...,  # default 0.0
```

Comment:
What's the reasoning behind the change here to `Any=...` with a comment of what the default was as opposed to keeping the original `Any=0.0` as per the default in the source?

Reply:
I was following [PEP 484](https://www.python.org/dev/peps/pep-0484/#id37).

Reply:
Looking at this more, this seems reasonable.

Reply:
Interestingly enough the PEP got changed recently, so these changes are a good way to deal with it.